strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/nuget/file/NuspecFile.java

```diff
@@ -0,0 +1,478 @@
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
+import javax.xml.transform.sax.SAXSource;
+
+import org.apache.commons.io.IOUtils;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+import org.xml.sax.XMLReader;
+import org.xml.sax.helpers.XMLReaderFactory;
+
+@XmlRootElement(
+        name = "package",
+        namespace = "http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd"
+)
+@XmlAccessorType(XmlAccessType.NONE)
+public class NuspecFile
+        implements Serializable
+{
+
+    public static final String NUSPEC_XML_NAMESPACE_2012 = "http://schemas.microsoft.com/packaging/2011/10/nuspec.xsd";
+    public static final String NUSPEC_XML_NAMESPACE_2013 = "http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd";
+    public static final String NUSPEC_XML_NAMESPACE_2016 = "http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd";
+    public static final String NUSPEC_XML_NAMESPACE_2013_01 = "http://schemas.microsoft.com/packaging/2013/01/nuspec.xsd";
+    public static final String NUSPEC_XML_NAMESPACE_2011 = "http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd";
+    public static final String NUSPEC_XML_NAMESPACE_2010 = "http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd";
+    public static final String NUSPEC_XML_NAMESPACE_2017 = "http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd";
+    public static final String NUSPEC_XML_NAMESPACE_EMPTY = "";
+    public static final String DEFAULT_FILE_EXTENSION = ".nuspec";
+    @XmlElement(
+            name = "metadata",
+            namespace = "http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd"
+    )
+    private NuspecFile.Metadata metadata;
+    @XmlElement(
+            name = "file",
+            namespace = "http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd"
+    )
+    @XmlElementWrapper(
+            name = "files",
+            namespace = "http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd"
+    )
+    private List<NuspecFile.NugetFile> files;
+
+    public static NuspecFile parse(InputStream inputStream)
+            throws NugetFormatException
+    {
+        try
+        {
+            JAXBContext context = JAXBContext.newInstance(NuspecFile.class);
+            Unmarshaller unmarshaller = context.createUnmarshaller();
+            unmarshaller.setEventHandler(new NuspecFile.NuspecXmlValidationEventHandler());
+            XMLReader reader = XMLReaderFactory.createXMLReader();
+            NugetNamespaceFilter inFilter = new NugetNamespaceFilter();
+            inFilter.setParent(reader);
+            InputSource inputSource = new InputSource(inputStream);
+            SAXSource saxSource = new SAXSource(inFilter, inputSource);
+            NuspecFile result = (NuspecFile) unmarshaller.unmarshal(saxSource);
+            return result;
+        }
+        catch (SAXException | JAXBException jaxbException)
```

Comment:
Cosmetic: Typically, in such cases, the exception would be called `e`, `exc`, or `exception`. Technically, it makes no difference, but this is less confusing when reading the code, (especially, when the list of thrown exceptions, is longer).